### ISSUE TITLE
Object indexing false positive

### ISSUE THREAD

**Issue by tankers746** [danielnixon/eslint-plugin-total-functions#26]
I'm wondering why the following example errors:
```
const periods = {
    oneDay: () => subDays(Date.now(), 1).valueOf(),
    oneWeek: () => subWeeks(Date.now(), 1).valueOf(),
    twoWeeks: () => subWeeks(Date.now(), 2).valueOf(),
    oneMonth: () => subMonths(Date.now(), 1).valueOf(),
    sixMonths: () => subMonths(Date.now(), 6).valueOf()
};
const period: keyof typeof periods = 'oneDay';
periods[period];
```

Great rule by the way, picking up lots of bugs.

**Comment by danielnixon** [comment]
Have you bumped into https://github.com/danielnixon/eslint-plugin-total-functions/issues/16 ?

**Comment by danielnixon** [comment]
https://github.com/danielnixon/eslint-plugin-total-functions/blob/master/src/rules/no-array-subscript.test.ts#L284-L296

**Comment by tankers746** [comment]
I have not yet, but would prefer that subscript access be split up into an array rule and an object rule. The rule for arrays, while not perfect, solves a pain point in typescript. I'm less concerned about object indexing which at least in my codebase is far more common than array indexing.

**Comment by danielnixon** [comment]
> would prefer that subscript access be split up into an array rule and an object rule

That's a fair point. PR welcome, otherwise I'll get to it soonish.

**Comment by danielnixon** [comment]
For the object and array decoupling: https://github.com/danielnixon/eslint-plugin-total-functions/issues/28

**Comment by danielnixon** [comment]
The explanation for this issue:

There's a limitation in the rule such that it won't consider non-literal keys (even if the key's type is `keyof typeof ...` or `const`).

So this won't (currently) pass the rule:
```typescript
const period: keyof typeof periods = 'oneDay';
periods[period];
```

but this will:
```
periods['oneDay'];
```

This is the object cousin of the array limitation in #16.

Happy to accept a PR, otherwise I'll fix this soonish too.

**Comment by danielnixon** [comment]
This might be fixed by the latest release. Please update and let me know @tankers746

**Comment by tankers746** [comment]
Just tested, works great!
